code/modules/integrated_electronics/subtypes/medical.dm

```diff
@@ -0,0 +1,668 @@
+	inputs = list(
+		"target"   = IC_PINTYPE_REF,
+		"organ"    = IC_PINTYPE_REF,
+		"bodypart" = IC_PINTYPE_STRING
+		)
+	surgery_items_type_list = list(
+		/obj/item/scalpel,
+		/obj/item/hemostat,
+		/obj/item/FixOVein,
+		/obj/item/organfixer,
+		/obj/item/organ/internal
+	)
+	spawn_flags = IC_SPAWN_RESEARCH
+	var/obj/item/organ/organ
+
+/obj/item/integrated_circuit/medical/surgery_device/internal/on_item_insert(obj/item/I)
+	..()
+	if(istype(I, /obj/item/scalpel))
+		operation_intent = SURGERY_ORGAN_DISCONNECT
+		st = new /datum/surgery_step/internal/detatch_organ()
+	else if(istype(I, /obj/item/hemostat))
+		operation_intent = SURGERY_ORGAN_REMOVE
+		st = new /datum/surgery_step/internal/remove_organ()
+	else if(istype(I, /obj/item/FixOVein))
+		operation_intent = SURGERY_ORGAN_CONNECT
+		st = new /datum/surgery_step/internal/attach_organ()
+	else if(istype(I, /obj/item/organfixer))
+		operation_intent = SURGERY_ORGAN_HEAL
+		st = new /datum/surgery_step/internal/fix_organ()
+	else
+		operation_intent = SURGERY_ORGAN_INSERT
+		st = new /datum/surgery_step/internal/replace_organ()
+
+/obj/item/integrated_circuit/medical/surgery_device/internal/do_work(ord)
+	var/mob/living/carbon/human/H = get_pin_data(IC_INPUT, 1)
+	var/obj/item/organ/internal/O = get_pin_data(IC_INPUT, 2)
+	var/obj/item/organ/external/E
+	if(!istype(H))
+		activate_pin(3)
+		return
+	selected_zone = get_pin_data(IC_INPUT, 3)
+	if(!istype(O))
+		activate_pin(3)
+		return
+	organ = O
+	if(!selected_zone || !(selected_zone in BP_ALL_LIMBS))
+		activate_pin(3)
+		return
+	if(selected_zone in H.op_stage.in_progress) //Can't operate on someone repeatedly.
+		activate_pin(3)
+		return
+	if(operation_intent == SURGERY_ORGAN_INSERT)
+		E = H.organs_by_name[O.parent_organ]
+	else
+		E = H.organs_by_name[selected_zone]
+	if(!istype(E))
+		activate_pin(3)
+		return
+
+	var/status = do_int_surgery(H)
+	if(status && !(status == SURGERY_FAILURE || status == SURGERY_BLOCKED))
+		organ = null
+		var/atom/A = get_object()
+		A.investigate_log("made some operation on ([H]) with [src].", INVESTIGATE_CIRCUIT)
+		activate_pin(2)
+	else
+		activate_pin(3)
+
+/obj/item/integrated_circuit/medical/surgery_device/internal/do_int_surgery(mob/living/carbon/M)
+	var/status = do_real_surgery(M, st, TRUE, organ)
+	if(status != SURGERY_FAILED_STATE)
+		return status
+	return FALSE
+
+/obj/item/integrated_circuit/medical/surgery_device/internal/can_use(mob/living/carbon/human/target, obj/item/organ/internal/organ, target_zone)
+	if(operation_intent == SURGERY_ORGAN_INSERT)
+		return st.can_use(src, target, target_zone, organ)
+	else
+		st.ignore_tool = TRUE
+		st.preselected_organ = organ
+		return st.can_use(src, target, target_zone, instrument)
+
+/obj/item/integrated_circuit/medical/surgery_device/face
+	name = "plastic surgery device"
+	desc = "This circuit contains instructions to use medical instruments for face manipulation. Perhaps it does operation like a surgery instrument inserted in it."
```

Comment:
```suggestion
	desc = "This device is a manipulator compatible with surgical instruments used for plastic surgery. While holding such an instrument, it can perform the facial reconstructing operation."
```